sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2835,6 +2835,84 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession {
     checkAnswer(df, Row(1, 3, 4) :: Row(2, 3, 4) :: Row(3, 3, 4) :: Nil)
   }
 
+  test("Support filter clause for aggregate function with hash") {
```

Comment:
what does "with hash" means here?

Reply:
Use `HashAggregateExec` or `ObjectHashAggregateExec`

Reply:
let's be explicit: `with hash aggregate`

Reply:
OK